vars/uploadMinSnapshotsToS3.groovy

```diff
@@ -68,9 +83,29 @@ void call(Map args = [:]) {
         string(credentialsId: 'jenkins-aws-production-account', variable: 'AWS_ACCOUNT_ARTIFACT'),
         string(credentialsId: 'jenkins-artifact-production-bucket-name', variable: 'ARTIFACT_PRODUCTION_BUCKET_NAME')]) {
             withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
+                // min artifacts
+                echo("Upload min snapshots")
                 s3Upload(file: "${srcDir}/${baseName}-latest.${extension}", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "${dstDir}/${baseName}-latest.${extension}")
                 s3Upload(file: "${srcDir}/${baseName}-latest.${extension}.sha512", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "${dstDir}/${baseName}-latest.${extension}.sha512")
                 s3Upload(file: "${srcDir}/${baseName}-latest.${extension}.build-manifest.yml", bucket: "${ARTIFACT_PRODUCTION_BUCKET_NAME}", path: "${dstDir}/${baseName}-latest.${extension}.build-manifest.yml")
+                // core plugins
+                if (architecture == "x64" && platform == "linux" && distribution == "tar" && corePluginDirExists) {
+                    echo("Upload core-plugins snapshots")
+                    List<String> corePluginList = buildManifestYamlOnly.components.artifacts.'core-plugins'[0]
```

Comment:
Why not retrieve component from already existing object `buildManifest`. We do have component class in it https://github.com/opensearch-project/opensearch-build-libraries/blob/main/src/jenkins/BuildManifest.groovy#L64

Reply:
This is a exact copy paste from promote artifact, we can improve these both later on.
Thanks.